### DISCUSSION TITLE
Should v6 warn or error when using the recommended-requiring-type-checking preset?

### DISCUSSION THREAD

**Discussion by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#6528] [RFCs]
### RFC

For context, in #6019 we discussed changing config names in v6. Part of that outcome is that the old `plugin:@typescript-eslint/recommended-requiring-type-checking` config would be renamed in v6 as `plugin:@typescript-eslint/recommended-type-checked`, with an alias left for easier backwards compat.

Quoting @JamesHenry's https://github.com/typescript-eslint/typescript-eslint/pull/6515#issuecomment-1443528204: 

> I know this is not technically related to this PR:
> 
> ![Screenshot of the v6 blog post indicating that the old recommended-requiring-type-checking blog post name is now an alias for recommended-type-checked; the alias will be removed in a future major version](https://user-images.githubusercontent.com/900523/221163930-5e6969fa-5ab8-45d8-9aae-d63c9446c81e.png)
> 
> ...but in reading through this I can't help thinking that it might be more desirable to be explicit and hard error when users try and use the old config names. If we provide a clear error message with a link to the docs outlining all the config options it could serve as a really useful prompt for users to revaluate their linting setups? (And a rare chance for us to do so in a major version). I can't help feeling like the pain of it going away will have to happen at some point, and if it happens now while it's fresh in everyone's minds we can make the experience as good as possible.
> 
> WDYT?

Now that I've had more time to ponder the breaking changes, I do like the idea of pressuring users to use the new config (and read the release notes). But I'm a little hesitant to hard break configs so immediately. How about we go with:

* v6: keep the alias, with a `console.warn`
* v7: throw an error
* v8: delete

cc @bradzacher

### Additional Info

Any additional info...

### Before you submit your RFC, please confirm the following. If any of these required steps are not taken, we may not be able to review your RFC. Help us to help you!

- [X] I have [searched for related discussions](https://github.com/typescript-eslint/typescript-eslint/discussions) and [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues) and found none that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

**Comment by bradzacher** [comment]
I don't think there's much difference between throwing an error and deleting it because in either case you'll get an error.

Part of me does dislike console warning because it breaks json based output - which would suck for users that can't control the transient usage of the name via configs.

**Reply by armano2** [reply]
agree, this is next major release that happens maybe once a year, and idea for major release that we can actually change stuff without backward compatibility
